cardano-api/src/Cardano/Api/Fees.hs

```diff
@@ -793,42 +794,62 @@ makeTransactionBodyAutoBalance eraInMode systemstart history pparams
     explicitTxFees <- first (const TxBodyErrorByronEraNotSupported) $
                         txFeesExplicitInEra era'
 
-    -- Insert change address and set tx fee to 0
+    -- Make a txbody that we will use for calculating the fees. For the purpose
+    -- of fees we just need to make a txbody of the right size in bytes. We do
+    -- not need the right values for the fee or change output. We use
+    -- "big enough" values for the change output and set so that the CBOR
+    -- encoding size of the tx will be big enough to cover the size of the final
+    -- output and fee. Yes this means this current code will only work for
+    -- final fee and change output of less than around 4000 ada (2^32-1 lovelace)
     txbody1 <- first TxBodyError $ -- TODO: impossible to fail now
                makeTransactionBody txbodycontent1 {
-                 txFee = TxFeeExplicit explicitTxFees 0
+                 txFee  = TxFeeExplicit explicitTxFees $ Lovelace (2^(32 :: Integer)),
```

Comment:
I guess this is set so that the this field takes up the maximal bytes for size calculation, right?

Reply:
Yes, though I think this is a hack and we've not understood something. I thought we would only need a 4 byte integer, i.e. 2^32-1. Jordan's example was then off by 2 bytes, and switching from 2^32-1 to 2^32 is basically a hack to add in an extra 4 bytes.

The fee cannot possibly be that high, indeed in Jordan's example the fee would actually fit into a 2 byte integer (i.e. it's less than 2^16-1).

It'd be preferable to track down where the missing bytes are. It looks like in Jordan's example we're actually off by exactly 4 bytes, since the validation error was for a fee for 2 bytes (2 * 44), and the fee field was already overly generous by 2 bytes.

Reply:
Also using 2^32 is a bit misleading. It'd be better to be more explicit that we're reserving 8 bytes by using 2^64-1